RecipeMatch/View Controllers/PreferencesViewController.m

```diff
@@ -13,121 +13,109 @@ @interface PreferencesViewController () <CCDropDownMenuDelegate>
     [titleView addSubview:title];
-
     self.navigationItem.titleView = titleView;
-
-    CGRect frame = CGRectMake((CGRectGetWidth(self.view.frame)-220), 133, 200, 37);
     
+    [self setupView];
+}
+
+- (void)setupView{
+    CGRect frame = CGRectMake((CGRectGetWidth(self.view.frame)-DROPDOWN_X_OFFSET), DROPDOWN_Y_POS, DROPDOWN_WIDTH, DROPDOWN_HEIGHT);
     self.cuisineMenu = [[ManaDropDownMenu alloc] initWithFrame:frame title:@"no preference"];
     self.cuisineMenu.numberOfRows = 19;
     self.cuisineMenu.textOfRows = @[@"American", @"Asian", @"British",@"Caribbean",@"Central Europe",@"Chinese", @"Eastern Europe", @"French", @"Indian", @"Italian", @"Japanese", @"Kosher", @"Mediterranean", @"Mexican", @"Middle Eastern", @"Nordic", @"South American", @"South East Asian", @"no preference"];
     self.cuisineMenu.activeColor = UIColorFromRGB(0x80CB99);
-    self.cuisineMenu.heightOfRows = 30;
+    self.cuisineMenu.heightOfRows = ROW_HEIGHT;
     self.cuisineMenu.delegate = self;
     [self.view addSubview:self.cuisineMenu];
     
-    
-    self.healthMenu = [[ManaDropDownMenu alloc] initWithFrame:CGRectOffset(frame, 0, 80) title:@"no preference"];
+    self.healthMenu = [[ManaDropDownMenu alloc] initWithFrame:CGRectOffset(frame, 0, MENU_OFFSET) title:@"no preference"];
     self.healthMenu.numberOfRows = 10;
     self.healthMenu.textOfRows = @[@"vegan", @"vegetarian", @"tree-nut-free",@"low-sugar",@"shellfish-free",@"pescatarian", @"paleo", @"gluten-free", @"fodmap-free", @"no preference"];
     self.healthMenu.activeColor = UIColorFromRGB(0x80CB99);
-    self.healthMenu.heightOfRows = 30;
+    self.healthMenu.heightOfRows = ROW_HEIGHT;
     self.healthMenu.delegate = self;
     [self.view addSubview:self.healthMenu];
 
-    self.dietMenu = [[ManaDropDownMenu alloc] initWithFrame:CGRectOffset(frame,0,160) title:@"no preference"];
+    self.dietMenu = [[ManaDropDownMenu alloc] initWithFrame:CGRectOffset(frame, 0, MENU_OFFSET * 2) title:@"no preference"];
     self.dietMenu.numberOfRows = 7;
     self.dietMenu.textOfRows = @[@"balanced", @"high-fiber", @"high-protein",@"low-carb",@"low-fat",@"low-sodium",@"no preference"];
     self.dietMenu.activeColor = UIColorFromRGB(0x80CB99);
-    self.dietMenu.heightOfRows = 30;
+    self.dietMenu.heightOfRows = ROW_HEIGHT;
     self.dietMenu.delegate = self;
     [self.view addSubview:self.dietMenu];
     
-    self.mealMenu = [[ManaDropDownMenu alloc] initWithFrame:CGRectOffset(frame, 0, 240) title:@"no preference"];
+    self.mealMenu = [[ManaDropDownMenu alloc] initWithFrame:CGRectOffset(frame, 0, MENU_OFFSET * 3) title:@"no preference"];
     self.mealMenu.numberOfRows = 6;
     self.mealMenu.textOfRows = @[@"breakfast", @"dinner", @"lunch",@"snack",@"teatime",@"no preference"];
     self.mealMenu.activeColor = UIColorFromRGB(0x80CB99);
-    self.mealMenu.heightOfRows = 30;
+    self.mealMenu.heightOfRows = ROW_HEIGHT;
     self.mealMenu.delegate = self;
     [self.view addSubview:self.mealMenu];
 }
```

Comment:
super stretch feature here😀!
add preference support on parse, and download it when opening the app.
that way, you won't have these static lists, and you'll be able to modify the  different preferences you support all from the cloud🙃

Reply:
ps `numberOfRows ` is always the size of the items you pass into `textOfRows `😉